### PR TITLE
Resolve SONARHTML-209 - Create rule S6807: DOM elements with ARIA roles should have the required properties

### PR DESCRIPTION
**Pull request by ericmorand-sonarsource**

### CODE REVIEW

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheck.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+public class ElementWithRoleShouldHaveRequiredPropertiesCheck extends AbstractPageCheck {
+  @Override
+  public void startElement(TagNode element) {
+    var roleName = element.getAttribute("role");
+    if (roleName == null) {
+      return;
+    }
+
+    var role = Aria.getRole(AriaRole.of(roleName));
+
```
Comment (zglicz) [inline sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheck.java:37]: please be consistent. Either remove here empty, or add 1 empty between roleName

File: sonar-html-plugin/src/test/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import org.sonar.plugins.html.visitor.HtmlSourceCode;
+
+import java.io.File;
+
+class ElementWithRoleShouldHaveRequiredPropertiesCheckTest {
+  @RegisterExtension
+  public CheckMessagesVerifierRule checkMessagesVerifier = new CheckMessagesVerifierRule();
+
+  @Test
+  void html() throws Exception {
```
Comment (zglicz) [inline sonar-html-plugin/src/test/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheckTest.java:33]: I'd split this test-case

**Review by zglicz — APPROVED** [review]
welldone

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheck.java (hunk 1 above)
Comment (ericmorand-sonarsource) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheck.java:37]: Good catch! Fixed.

File: sonar-html-plugin/src/test/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheckTest.java (hunk 2 above)
Comment (ericmorand-sonarsource) [reply sonar-html-plugin/src/test/java/org/sonar/plugins/html/checks/accessibility/ElementWithRoleShouldHaveRequiredPropertiesCheckTest.java:33]: Done, thanks.

**Pull request merged**